test/built-ins/Array/prototype/findIndex/return-abrupt-from-this-length.js

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 22.1.3.9
+description: >
+  Return abrupt from ToLength(Get(O, "length")).
+info: >
+  22.1.3.9 Array.prototype.findIndex ( predicate[ , thisArg ] )
+
+  1. Let O be ToObject(this value).
+  2. ReturnIfAbrupt(O).
+  3. Let len be ToLength(Get(O, "length")).
+  4. ReturnIfAbrupt(len).
+---*/
+
+var o = {};
+
+Object.defineProperty(o, 'length', {
+  get: function() {
+    throw new Test262Error();
+  },
+  configurable: true
+});
+// predicate fn is given to avoid false positives
+assert.throws(Test262Error, function() {
+  [].findIndex.call(o, function() {});
+});
+
+delete o.length;
```

Comment:
You can simplify this test a little by just making a new object to use as the "this" value for the second invocation. Might be slightly safer, too (since it avoids shared state)